projects/rocprofiler-systems/source/lib/rocprof-sys/library/amd_smi/settings.hpp

```diff
@@ -0,0 +1,174 @@
+#pragma once
+
+#include "core/config.hpp"
+#include "core/debug.hpp"
+#include "library/amd_smi/common.hpp"
+#include <regex.h>
+
+namespace rocprofsys
+{
+namespace amd_smi
+{
+
+namespace
+{
+constexpr uint16_t enable_all_metrics  = 0xffff;
+constexpr uint16_t disable_all_metrics = 0x0000;
+
+}  // namespace
+
+struct settings
+{
+    static device_filter get_device_filter()
+    {
+        auto filter = rocprofsys::get_sampling_gpus();
+        if(filter == "all" || filter == "on")
+        {
+            return { .mode = device_selection_mode::all, .indices = {} };
+        }
+
+        if(filter == "none" || filter == "off")
+        {
+            return { .mode = device_selection_mode::none, .indices = {} };
+        }
+
+        auto enabled_devices = parse_numeric_range(filter);
+
+        return { .mode = device_selection_mode::specific, .indices = enabled_devices };
+    }
+
+    static smi_metric_options get_enabled_metrics()
+    {
+        static auto _enabled_metrics = [] {
+            auto settings = get_setting_value<std::string>("ROCPROFSYS_AMD_SMI_METRICS");
+            return parse_enabled_metrics(settings.has_value() ? settings.value() : "");
+        }();
+        return _enabled_metrics;
+    };
```

Comment:
Did you considered to make it like:

```suggestion
    static smi_metric_options get_enabled_metrics(std::string _settings)
    {
        static auto _enabled_metrics = [](std::string_view settings) {
            return parse_enabled_metrics(settings.has_value() ? settings.value() : "");
        }(_settings);
        return _enabled_metrics;
    };
```

We will get rid of env. dependency and decouple it from env.
We can figure out new way to cache settings, to avoid processing it each time.